### PR TITLE
Add cancel button on user number prompt

### PR DESCRIPTION
**Pull request by nmattia**
... So that the user can cancel.

### CODE REVIEW

File: src/frontend/src/flows/recovery/useRecovery.ts [hunk 1]
```diff
@@ -8,10 +8,20 @@ import { deviceRecoveryPage } from "./recoverWith/device";
...
 export const useRecovery = async (userNumber?: bigint): Promise<void> => {
-  userNumber =
-    userNumber === undefined
-      ? await promptUserNumber("Recover Identity Anchor", null)
-      : userNumber;
+  if (userNumber !== undefined) {
+    return runRecovery(userNumber);
+  } else {
+    const pUserNumber = await promptUserNumber("Recover Identity Anchor", null);
+    console.log(pUserNumber);
```
Comment (frederikrothenberger) [inline src/frontend/src/flows/recovery/useRecovery.ts:15]: Nit: remove?

**Review by frederikrothenberger — APPROVED** [review]
LGTM!

File: src/frontend/src/flows/recovery/useRecovery.ts (hunk 1 above)
Comment (nmattia) [reply src/frontend/src/flows/recovery/useRecovery.ts:15]: Woops! Will fix. Auto-merge was on.

Comment (nmattia) [reply src/frontend/src/flows/recovery/useRecovery.ts:15]: https://github.com/dfinity/internet-identity/pull/646

**Pull request merged**